packages/reputation-miner/ReputationMiner.js

```diff
@@ -758,12 +761,131 @@ class ReputationMiner {
+      if (res.length === 0) {
+        res = await db.run(`INSERT INTO skills (skill_id) VALUES ('${skillId}')`); // eslint-disable-line no-await-in-loop
+        skillDBId = res.lastID;
+      } else {
+        skillDBId = res[0].skill_id;
+      }
+
+      // eslint-disable-next-line no-await-in-loop
+      res = await db.get(
+        `SELECT COUNT ( * ) AS "n"
+        FROM reputations
+        WHERE root_hash=${rootHashDBId}
+        AND colony_address=${colonyDBId}
+        AND skill_id=${skillDBId}
+        AND user_address=${userDBId}`
+      );
+
+      if (res.n === 0) {
+        // eslint-disable-next-line no-await-in-loop
+        await db.run(
+          `INSERT INTO reputations (root_hash, colony_address, skill_id, user_address, value)
+          VALUES (${rootHashDBId}, ${colonyDBId}, ${skillDBId}, ${userDBId}, '${value}')`
+        );
+      }
+    }
+    await db.close();
+  }
+
+  async loadState(reputationRootHash) {
+    const db = await sqlite.open(this.dbPath, { Promise });
+    this.nReputations = ethers.utils.bigNumberify(0);
+    this.reputations = {};
+
+    const res = await db.all(
+      `SELECT reputations.skill_id, reputations.value, reputation_states.root_hash, colonies.address as colony_address, users.address as user_address
+       FROM reputations
+       INNER JOIN colonies ON colonies.rowid=reputations.colony_address
+       INNER JOIN users ON users.rowid=reputations.user_address
+       INNER JOIN reputation_states ON reputation_states.rowid=reputations.root_hash
+       WHERE reputation_states.root_hash="${reputationRootHash}"`
+    );
+    this.nReputations = ethers.utils.bigNumberify(res.length);
+    for (let i = 0; i < res.length; i += 1) {
+      const row = res[i];
+      const key = await ReputationMiner.getKey(row.colony_address, row.skill_id, row.user_address); // eslint-disable-line no-await-in-loop
+      await this.reputationTree.insert(key, row.value, { gasLimit: 4000000 }); // eslint-disable-line no-await-in-loop
+      this.reputations[key] = row.value;
+    }
+    await db.close();
+  }
+
+  async resetDB() {
+    const db = await sqlite.open(this.dbPath, { Promise });
+    await db.run(`DROP TABLE IF EXISTS users`);
+    await db.run(`DROP TABLE IF EXISTS colonies`);
+    await db.run(`DROP TABLE IF EXISTS skills`);
+    await db.run(`DROP TABLE IF EXISTS reputations`);
+    await db.run(`DROP TABLE IF EXISTS reputation_states`);
+    await db.run("CREATE TABLE users ( address text NOT NULL UNIQUE )");
+    await db.run("CREATE TABLE reputation_states ( root_hash text NOT NULL UNIQUE, n_nodes INTEGER NOT NULL )");
+    await db.run("CREATE TABLE colonies ( address text NOT NULL UNIQUE )");
+    await db.run("CREATE TABLE skills ( skill_id INTEGER PRIMARY KEY )");
+    await db.run(
+      `CREATE TABLE reputations (
+        root_hash text NOT NULL,
+        colony_address INTEGER NOT NULL,
+        skill_id INTEGER NOT NULL,
+        user_address INTEGER NOT NULL,
```

Comment:
The column namings confused me at first when I saw them for `root_hash`, `colony_address`, `skill_id` and `user_address` as those are not holding respectively a root hash, address, skill id or an address but contain sql table identifiers. We should name them perhaps postfix them with `dbId` and remove `address` from the names? e.g. `root_hash_dbId`, `colony_dbId`, `skill_dbId` and `user_dbId?

Reply:
Postfixed with `rowid`, as that is the name of the column that holds these keys in the tables where they are the primary keys.